aws/resource_aws_service_discovery_public_dns_namespace.go

```diff
@@ -0,0 +1,144 @@
+package aws
+
+import (
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/servicediscovery"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsServiceDiscoveryPublicDnsNamespace() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsServiceDiscoveryPublicDnsNamespaceCreate,
+		Read:   resourceAwsServiceDiscoveryPublicDnsNamespaceRead,
+		Delete: resourceAwsServiceDiscoveryPublicDnsNamespaceDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"hosted_zone": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsServiceDiscoveryPublicDnsNamespaceCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sdconn
+
+	input := &servicediscovery.CreatePublicDnsNamespaceInput{
+		Name: aws.String(d.Get("name").(string)),
+	}
+
+	if v, ok := d.GetOk("description"); ok {
+		input.Description = aws.String(v.(string))
+	}
+
+	resp, err := conn.CreatePublicDnsNamespace(input)
+	if err != nil {
+		return err
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{servicediscovery.OperationStatusSubmitted, servicediscovery.OperationStatusPending},
+		Target:     []string{servicediscovery.OperationStatusSuccess},
+		Refresh:    servicediscoveryOperationRefreshStatusFunc(conn, *resp.OperationId),
+		Timeout:    5 * time.Minute,
+		Delay:      10 * time.Second,
+		MinTimeout: 3 * time.Second,
```

Comment:
Is there any particular reason for delaying the refresh here? i.e. do the two fields `Delay` & `MinTimeout` have any reason? Based on my brief testing the creation can finish in 2 seconds, so we're potentially letting  the user wait for 8 more seconds unnecessarily here.